client/client.go

```diff
@@ -95,14 +189,21 @@ func (client *Client) writeLoop() {
 	// Pipeline requests; but only write them one at a time. To allow multiple
 	// goroutines to all write as quickly as possible, uses a channel and the
 	// writeLoop lives in a separate goroutine.
-	for req := range client.outbound {
-		client.conn.Write([]byte(rt.ReqStr))
+	for req := range client.channels.outbound {
 
-		// todo handle errors.
+		cleanupHandle := func() {
```

Comment:
This closure doesn't seem necessary. You could just pass the `req` into the `writeReconnectCleanup` method and rather than calling the closure just call `client.requestPool.put(req)` at that point.